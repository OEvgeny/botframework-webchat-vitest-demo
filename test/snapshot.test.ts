import { expect, test } from 'vitest';
import 'botframework-webchat/dist/webchat.js';

test('snapshot', () => {
  expect(WebChat).toMatchSnapshot();
});
