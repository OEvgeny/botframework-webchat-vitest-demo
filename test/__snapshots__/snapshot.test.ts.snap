// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot 1`] = `
{
  "Components": {
    "AccessKeySinkSurface": [Function],
    "AdaptiveCardContent": [Function],
    "AnimationCardContent": [Function],
    "AudioCardContent": [Function],
    "AudioContent": [Function],
    "Avatar": [Function],
    "BasicConnectivityStatus": [Function],
    "BasicSendBox": [Function],
    "BasicToaster": [Function],
    "BasicTranscript": [Function],
    "BasicWebChat": [Function],
    "Bubble": [Function],
    "Composer": [Function],
    "DictationInterims": [Function],
    "ErrorBox": [Function],
    "FileContent": [Function],
    "HTMLVideoContent": [Function],
    "HeroCardContent": [Function],
    "ImageContent": [Function],
    "MicrophoneButton": [Function],
    "OAuthCardContent": [Function],
    "ReceiptCardContent": [Function],
    "SendButton": [Function],
    "SendStatus": [Function],
    "SendTextBox": [Function],
    "SignInCardContent": [Function],
    "SpeakActivity": [Function],
    "SuggestedActions": [Function],
    "TextContent": {
      "$$typeof": Symbol(react.memo),
      "compare": null,
      "defaultProps": {
        "contentType": "text/plain",
      },
      "displayName": "TextContent",
      "type": [Function],
    },
    "ThumbnailCardContent": [Function],
    "Timestamp": [Function],
    "UploadButton": [Function],
    "VideoCardContent": [Function],
    "VideoContent": [Function],
    "VimeoContent": [Function],
    "YouTubeContent": [Function],
    "connectDictationInterims": [Function],
    "connectMicrophoneButton": [Function],
    "connectSendButton": [Function],
    "connectSendStatus": [Function],
    "connectSpeakActivity": [Function],
    "connectSuggestedActions": [Function],
    "connectUploadButton": [Function],
  },
  "Constants": {
    "ActivityClientState": {
      "SENDING": "sending",
      "SEND_FAILED": "send failed",
      "SENT": "sent",
    },
    "DictateState": {
      "DICTATING": 3,
      "IDLE": 0,
      "STARTING": 2,
      "STOPPING": 4,
      "WILL_START": 1,
    },
  },
  "ReactWebChat": [Function],
  "concatMiddleware": [Function],
  "connectToWebChat": [Function],
  "createAdaptiveCardsAttachmentForScreenReaderMiddleware": [Function],
  "createAdaptiveCardsAttachmentMiddleware": [Function],
  "createBrowserWebSpeechPonyfillFactory": [Function],
  "createCognitiveServicesSpeechServicesPonyfillFactory": [Function],
  "createDirectLine": [Function],
  "createDirectLineAppServiceExtension": [Function],
  "createDirectLineSpeechAdapters": [Function],
  "createStore": [Function],
  "createStoreWithOptions": [Function],
  "createStyleSet": [Function],
  "hooks": {
    "useActiveTyping": [Function],
    "useActivities": [Function],
    "useActivityKeys": [Function],
    "useActivityKeysByRead": [Function],
    "useAdaptiveCardsHostConfig": [Function],
    "useAdaptiveCardsPackage": [Function],
    "useAvatarForBot": [Function],
    "useAvatarForUser": [Function],
    "useByteFormatter": [Function],
    "useConnectivityStatus": [Function],
    "useCreateActivityRenderer": [Function],
    "useCreateActivityStatusRenderer": [Function],
    "useCreateAttachmentForScreenReaderRenderer": [Function],
    "useCreateAvatarRenderer": [Function],
    "useCreateScrollToEndButtonRenderer": [Function],
    "useDateFormatter": [Function],
    "useDebouncedNotifications": [Function],
    "useDictateAbortable": [Function],
    "useDictateInterims": [Function],
    "useDictateState": [Function],
    "useDirection": [Function],
    "useDisabled": [Function],
    "useDismissNotification": [Function],
    "useEmitTypingIndicator": [Function],
    "useFocus": [Function],
    "useFocusSendBox": [Function],
    "useGetActivityByKey": [Function],
    "useGetHasAcknowledgedByActivityKey": [Function],
    "useGetKeyByActivity": [Function],
    "useGetKeyByActivityId": [Function],
    "useGetSendTimeoutForActivity": [Function],
    "useGrammars": [Function],
    "useGroupActivities": [Function],
    "useGroupTimestamp": [Function],
    "useLanguage": [Function],
    "useLastAcknowledgedActivityKey": [Function],
    "useLastReadActivityKey": [Function],
    "useLocalizer": [Function],
    "useMarkActivityAsSpoken": [Function],
    "useMarkActivityKeyAsRead": [Function],
    "useMarkAllAsAcknowledged": [Function],
    "useMicrophoneButtonClick": [Function],
    "useMicrophoneButtonDisabled": [Function],
    "useObserveScrollPosition": [Function],
    "useObserveTranscriptFocus": [Function],
    "usePerformCardAction": [Function],
    "usePonyfill": [Function],
    "usePostActivity": [Function],
    "useReferenceGrammarID": [Function],
    "useRelativeTimeFormatter": [Function],
    "useRenderActivity": [Function],
    "useRenderActivityStatus": [Function],
    "useRenderAttachment": [Function],
    "useRenderAvatar": [Function],
    "useRenderMarkdownAsHTML": [Function],
    "useRenderToast": [Function],
    "useRenderTypingIndicator": [Function],
    "useScrollDown": [Function],
    "useScrollTo": [Function],
    "useScrollToEnd": [Function],
    "useScrollUp": [Function],
    "useSendBoxSpeechInterimsVisible": [Function],
    "useSendBoxValue": [Function],
    "useSendEvent": [Function],
    "useSendFiles": [Function],
    "useSendMessage": [Function],
    "useSendMessageBack": [Function],
    "useSendPostBack": [Function],
    "useSendStatusByActivityKey": [Function],
    "useSendTimeoutForActivity": [Function],
    "useSendTypingIndicator": [Function],
    "useSetNotification": [Function],
    "useShouldSpeakIncomingActivity": [Function],
    "useStartDictate": [Function],
    "useStopDictate": [Function],
    "useStyleOptions": [Function],
    "useStyleSet": [Function],
    "useSubmitSendBox": [Function],
    "useSuggestedActions": [Function],
    "useTextBoxSubmit": [Function],
    "useTextBoxValue": [Function],
    "useTimeoutForSend": [Function],
    "useTrackDimension": [Function],
    "useTrackEvent": [Function],
    "useTrackException": [Function],
    "useTrackTiming": [Function],
    "useTypingIndicatorVisible": [Function],
    "useUserID": [Function],
    "useUsername": [Function],
    "useVoiceSelector": [Function],
    "useWebSpeechPonyfill": [Function],
  },
  "renderMarkdown": [Function],
  "renderWebChat": [Function],
  "withEmoji": [Function],
}
`;
